### PR TITLE
[MBUILDCACHE-96] Remove use of maven-compat

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: Olivier Lamy <olamy@apache.org>

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [ ] Make sure there is a [MBUILDCACHE JIRA issue](https://issues.apache.org/jira/browse/MBUILDCACHE) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[MBUILDCACHE-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MBUILDCACHE-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [ ] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [ ] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

File: src/main/java/org/apache/maven/buildcache/checksum/MavenProjectInput.java [hunk 1]
```diff
@@ -684,51 +804,58 @@ private SortedMap<String, String> getMutableDependenciesHashes(String keyPrefix,
...
+        org.eclipse.aether.artifact.Artifact dependencyArtifact = new org.eclipse.aether.artifact.DefaultArtifact(
+                dependency.getGroupId(),
+                dependency.getArtifactId(),
+                dependency.getClassifier(),
+                null,
+                dependency.getVersion(),
+                new DefaultArtifactType(dependency.getType()));
+        ArtifactRequest artifactRequest = new ArtifactRequest().setArtifact(dependencyArtifact);
+
+        ArtifactResult result = artifactResolver.resolveArtifact(session.getRepositorySession(), artifactRequest);
```
Comment (cstamas) [inline src/main/java/org/apache/maven/buildcache/checksum/MavenProjectInput.java:834]: -1, this is `impl` package, use public API instead instead:
https://github.com/apache/maven-resolver/blob/master/maven-resolver-api/src/main/java/org/eclipse/aether/RepositorySystem.java#L164-L178

Comment (olamy) [reply src/main/java/org/apache/maven/buildcache/checksum/MavenProjectInput.java]: fixed

File: src/main/java/org/apache/maven/buildcache/DefaultProjectInputCalculator.java [hunk 2]
```diff
@@ -27,13 +27,15 @@
...
+import org.apache.maven.artifact.handler.manager.ArtifactHandlerManager;
 import org.apache.maven.buildcache.checksum.MavenProjectInput;
 import org.apache.maven.buildcache.xml.CacheConfig;
 import org.apache.maven.buildcache.xml.build.ProjectsInputInfo;
 import org.apache.maven.execution.MavenSession;
 import org.apache.maven.lifecycle.internal.builder.BuilderCommon;
 import org.apache.maven.project.MavenProject;
-import org.apache.maven.repository.RepositorySystem;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.impl.ArtifactResolver;
```
Comment (cstamas) [inline src/main/java/org/apache/maven/buildcache/DefaultProjectInputCalculator.java:38]: Remove this pls

Comment (olamy) [reply src/main/java/org/apache/maven/buildcache/DefaultProjectInputCalculator.java]: fixed

File: src/main/java/org/apache/maven/buildcache/checksum/MavenProjectInput.java [hunk 3]
```diff
@@ -684,51 +801,58 @@ private SortedMap<String, String> getMutableDependenciesHashes(String keyPrefix,
...
-                .setForceUpdate(session.getRequest().isUpdateSnapshots())
-                .setServers(session.getRequest().getServers())
-                .setMirrors(session.getRequest().getMirrors())
-                .setProxies(session.getRequest().getProxies());
-
-        final ArtifactResolutionResult result = repoSystem.resolve(request);
-
-        if (!result.isSuccess()) {
+    private DigestItem resolveArtifact(final Dependency dependency)
+            throws IOException, ArtifactResolutionException, InvalidVersionSpecificationException {
```
[the remark this answers is not in the document]
Comment (cstamas) [reply src/main/java/org/apache/maven/buildcache/checksum/MavenProjectInput.java:820]: Ah, i see..., nvm

**Review by cstamas — APPROVED** [review]

**Comment by cstamas** [comment]
Perso, I'd not mix maven-artifact and resolver Artifact/Version stuff, but that is not related to PR goal, so might be done in some subsequent PR.

**Comment by jira-importer** [comment]
Resolve #303

**Comment by jira-importer** [comment]
Resolve #303

**Pull request merged**